src/Message.php

```diff
@@ -16,7 +16,7 @@
     public function __construct($payload, $timestamp, $encoding)
     {
         $this->payload = $payload;
-        $this->timestamp = new \DateTimeImmutable($timestamp);
+        $this->timestamp = new \DateTimeImmutable("@$timestamp");
```

Comment:
The thing is that `AMQPMessage` already uses the UNIX timestamps, and we are just transforming this to the `DateTimeImmutable` in Reagieren's `Message` object, not much to do in here I guess.